Storage/src/Connection/Rest.php

```diff
@@ -628,6 +629,157 @@ private function sendWithRetry($resource, $method, $args)
             $args
         );
 
+        $requestHash = Uuid::uuid4()->toString();
+        $args['onRetryException'] = function (
+            \Exception $e,
+            $currentAttempt,
+            &$arguments
+        ) use ($requestHash) {
+            // Since we the the last attempt number here, so incrementing it
+            // to get the current attempt count
+            ++$currentAttempt;
+            $this->updateRetryInovationHeaders(
+                $arguments,
+                $requestHash,
+                $currentAttempt
+            );
+        };
+
+        $args['onExecutionStart'] = function (&$arguments) use ($requestHash) {
+            $this->updateRetryInovationHeaders(
+                $arguments,
+                $requestHash
+            );
+        };
+
         return $this->send($resource, $method, $args);
     }
+
+    /**
+     * Updates the x-goog-api-client header value with UUID
+     * and retry count
+     *
+     * @return void
+     */
+    private function updateRetryInovationHeaders(
+        &$arguments,
+        $requestHash,
+        $currentAttempt = 0
+    ) {
+        // Retrieve $request and $options from $arguments
+        foreach ($arguments as &$argument) {
+            if ($argument instanceof Request) {
+                $request = $argument;
+            } elseif (isset($argument['headers'])) {
+                $options = &$argument;
+            }
+        }
```

Comment:
Hasn't this been abstracted out into `fetchRequest` and `fetchOptions`?

Reply:
Yeah, that piece of code was mistakenly left over. Removed them.

[2014c4c](https://github.com/googleapis/google-cloud-php/pull/5741/commits/2014c4c77cbde0ccca4e7b3ff7b041301f5cf01b)